fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java

```diff
@@ -670,24 +669,25 @@ protected DataSourceMetadataLoader buildDataSourceMetadataLoader(
     }
 
     /**
-     * Build a DruidDimensionsLoader.
+     * Build a DimensionLoader.
      *
      * @param webService  The web service used by the loader to query dimension values
      * @param physicalTableDictionary  The table to update dimensions on
      * @param dimensionDictionary  The dimensions to update
      *
-     * @return A DruidDimensionsLoader
+     * @return A DimensionLoader
      */
-    protected DruidDimensionsLoader buildDruidDimensionsLoader(
+    protected DimensionLoader buildDruidDimensionsLoader(
```

Comment:
I could mark both `buildDruidDimensionsLoader(...)` and `setupDruidDimensionsLoader(...)` as deprecated and make a method to provide the collection of `AbstractDimensionRowProvider`?